### ISSUE TITLE
Improve script search performance

### ISSUE THREAD

**Issue by tchajed** [tchajed/botc-tools#15]
There are now enough scripts that search is actually somewhat slow, especially with the character search feature. I put no effort into performance in the first implementation. It's now worth it to look into performance, perhaps by adding some full-text indexing of script titles (with a library) or by doing something less naive (like search a list of characters, then search an index of scripts by character). It might also be useful to cancel an existing search if possible.

**Comment by jthemphill** [comment]
Here's a very fast one that indexes all the scripts as a Webassembly module at build time: https://stork-search.net/

**Comment by jthemphill** [comment]
https://www.fusejs.io/ looks like a good drop-in replacement, I'll try this

**Comment by tchajed** [comment]
Yeah that looks great - we'll have both fuzzy searching and character + title search, which is perfect. Looks like a little work is needed to generate an index at build time, but you should be able to integrate it into fetch-assets.
